bgpd/bgp_zebra.c

```diff
@@ -793,12 +794,20 @@ bool bgp_zebra_nexthop_set(union sockunion *local, union sockunion *remote,
 			 * route-map to
 			 * specify the global IPv6 nexthop.
 			 */
-			if_get_ipv6_local(ifp, &nexthop->v6_global);
+			v6_ll_avail =
+				if_get_ipv6_local(ifp, &nexthop->v6_global);
 			memcpy(&nexthop->v6_local, &nexthop->v6_global,
 			       IPV6_MAX_BYTELEN);
 		} else
-			if_get_ipv6_local(ifp, &nexthop->v6_local);
+			v6_ll_avail =
+				if_get_ipv6_local(ifp, &nexthop->v6_local);
 
+		/*
+		 * If we are a v4 connection and we are not doing unnumbered
+		 * not having a v6 LL address is ok
+		 */
+		if (!v6_ll_avail && !peer->conf_if)
+			v6_ll_avail = true;
```

Comment:
I think this breaks IPv4-only sessions if there's an interface configured, e.g. `neighbor 1.2.3.4 interface eth0` without IPv6 on eth0.

Both of these blocks should probably only be executed if the "other" address family is actually enabled on the peer (which wasn't a problem before but is after this.)

Reply:
Nevermind, confused `neighbor IFACE` with `neighbor A.B.C.D interface IFACE`

Reply:
so `neighbor 1.2.3.4 interface eth0` does not store the interface in `peer->conf_if`.